### PR TITLE
I sonny api create and view pariwisata

### PR DESCRIPTION
**Pull request by sonnyhutabarat**

### CODE REVIEW

File: pkg/delivery/pariwisata/http/init.go [hunk 1]
```diff
@@ -17,6 +17,7 @@ func InitPariwisataHandler(e *echo.Echo, p pariwisataUsecase.Usecase) {
 
 	//register handler
 	e.GET("/pariwisata", handler.GetAllPariwisata)
+	e.POST("/create", handler.CreatePariwisata)
```
Comment (geshtng) [inline pkg/delivery/pariwisata/http/init.go]:
```suggestion
	e.POST("api/pariwisata/create", handler.CreatePariwisata)
```

File: pkg/delivery/pariwisata/http/pariwisata.go [hunk 2]
```diff
@@ -22,3 +22,27 @@ func (d *pariwisata) GetAllPariwisata(c echo.Context) error {
 	c.Response().Header().Set(`X-Cursor`, "header")
 	return c.JSON(http.StatusOK, allPariwisata)
 }
+
+//for Insert wisata
+func (d *pariwisata) CreatePariwisata(c echo.Context) error {
+
+	// pName := c.QueryParam("name")
```
Comment (geshtng) [inline pkg/delivery/pariwisata/http/pariwisata.go]: Delete

File: pkg/delivery/pariwisata/http/pariwisata.go [hunk 3]
```diff
@@ -22,3 +22,27 @@ func (d *pariwisata) GetAllPariwisata(c echo.Context) error {
...
+	// pLokasi := c.QueryParam("lokasi")
```
Comment (geshtng) [inline pkg/delivery/pariwisata/http/pariwisata.go]: Delete

File: pkg/repository/pariwisata/postgres/pariwisata.go [hunk 4]
```diff
@@ -26,3 +26,43 @@ func (r *pariwisata) GetAllPariwisata() ([]models.Pariwisata, error) {
...
+// 	// 	pariwisata := models.Pariwisata{}
+// 	// 	err := rows.Scan(&pariwisata.ID, &pariwisata.Name, &pariwisata.Lokasi)
+// 	// 	if err != nil {
+// 	// 		log.Println(err)
+// 	// 	}
+// 	// 	allPariwisata = append(allPariwisata, pariwisata)
+// 	// }
+
+// 	// return allPariwisata, nil
+// }
```
Comment (geshtng) [inline pkg/repository/pariwisata/postgres/pariwisata.go]: Delete all commented code

File: pkg/repository/pariwisata/postgres/statement.go [hunk 5]
```diff
@@ -1,3 +1,5 @@
 package postgres
 
 var queryGetAllPariwisata = "SELECT pariwisata_id, pariwisata_name, pariwisata_lokasi FROM pariwisata"
+
+var queryCreatePariwisata = `INSERT INTO pariwisata (pariwisata_name, pariwisata_lokasi)VALUES($1, $2) RETURNING pariwisata_id`
```
Comment (geshtng) [inline pkg/repository/pariwisata/postgres/statement.go]:
```suggestion
var queryCreatePariwisata = `INSERT INTO pariwisata (pariwisata_name, pariwisata_lokasi) VALUES ($1, $2)`
```

File: pkg/repository/pariwisata/repository.go [hunk 6]
```diff
@@ -6,4 +6,5 @@ import (
 
 type Repository interface {
 	GetAllPariwisata() ([]models.Pariwisata, error)
+	CreatePariwisata(pariwisataName, pariwisataLokasi string) error
```
Comment (geshtng) [inline pkg/repository/pariwisata/repository.go]:
```suggestion
	CreatePariwisata(pariwisataName string, pariwisataLokasi string) error
```

File: pkg/usecase/pariwisata/module/pariwisata.go [hunk 7]
```diff
@@ -1,6 +1,8 @@
 package module
 
 import (
+
+	// "log"
```
Comment (geshtng) [inline pkg/usecase/pariwisata/module/pariwisata.go]: Delete

File: pkg/usecase/pariwisata/module/pariwisata.go [hunk 8]
```diff
@@ -12,3 +14,25 @@ func (u *pariwisata) GetAllPariwisata() ([]models.Pariwisata, error) {
...
+
+//for the insert
+// func (u *pariwisata) CreatePariwisata() ([]models.Pariwisata, error) {
+// 	allPariwisata, err := u.pariwisataRepo.CreatePariwisata()
+// 	// if err != nil {
+// 	// 	log.Println(err)
+// 	// }
+
+// 	// return allPariwisata, err
+// }
```
Comment (geshtng) [inline pkg/usecase/pariwisata/module/pariwisata.go]: Delete commented code

File: pkg/usecase/pariwisata/module/pariwisata.go [hunk 9]
```diff
@@ -12,3 +14,25 @@ func (u *pariwisata) GetAllPariwisata() ([]models.Pariwisata, error) {
...
+
+func (u *pariwisata) CreatePariwisata(pName, pLokasi string) error {
+	// allPariwisata, err := u.pariwisataRepo.CreatePariwisata()
+	// if err != nil {
+	// 	log.Println(err)
+	// }
```
Comment (geshtng) [inline pkg/usecase/pariwisata/module/pariwisata.go]: Delete commented code

File: pkg/usecase/pariwisata/module/pariwisata.go [hunk 10]
```diff
@@ -12,3 +14,25 @@ func (u *pariwisata) GetAllPariwisata() ([]models.Pariwisata, error) {
...
+	err := u.pariwisataRepo.CreatePariwisata(pName, pLokasi)
+
+	// log.Println(pName)
```
Comment (geshtng) [inline pkg/usecase/pariwisata/module/pariwisata.go]: Delete

File: pkg/delivery/pariwisata/http/pariwisata.go [hunk 11]
```diff
@@ -22,3 +22,27 @@ func (d *pariwisata) GetAllPariwisata(c echo.Context) error {
 	c.Response().Header().Set(`X-Cursor`, "header")
 	return c.JSON(http.StatusOK, allPariwisata)
 }
+
+//for Insert wisata
```
Comment (geshtng) [inline pkg/delivery/pariwisata/http/pariwisata.go]:
```suggestion
// CreatePariwisata for inserting new data to pariwisata
```

File: pkg/delivery/pariwisata/http/pariwisata.go [hunk 12]
```diff
@@ -15,10 +15,36 @@ func (d *pariwisata) GetAllPariwisata(c echo.Context) error {
 	}
 
 	allPariwisata, err := d.pariwisataUsecase.GetAllPariwisata()
+	if err != nil {
+		log.Println("[Delivery][Pariwisata][GetAllPariwisata] Error : ", err)
+
+		return c.JSON(http.StatusInternalServerError, allPariwisata)
+	}
+
+	c.Response().Header().Set(`Content-Type`, "application/json")
```
Comment (geshtng) [inline pkg/delivery/pariwisata/http/pariwisata.go]:
```suggestion
	c.Response().Header().Set("Content-Type", "application/json")
```

File: pkg/delivery/pariwisata/http/pariwisata.go [hunk 13]
```diff
@@ -15,10 +15,36 @@ func (d *pariwisata) GetAllPariwisata(c echo.Context) error {
...
+	pLokasi := c.FormValue("lokasi")
+
+	ctx := c.Request().Context()
+	if ctx == nil {
+		ctx = context.Background()
+	}
+
+	err := d.pariwisataUsecase.CreatePariwisata(pName, pLokasi)
 	if err != nil {
 		log.Println(err)
```
Comment (geshtng) [inline pkg/delivery/pariwisata/http/pariwisata.go]: Improve logging

File: pkg/delivery/pariwisata/http/pariwisata.go [hunk 14]
```diff
@@ -15,10 +15,36 @@ func (d *pariwisata) GetAllPariwisata(c echo.Context) error {
...
 	}
 
 	c.Response().Header().Set(`X-Cursor`, "header")
-	return c.JSON(http.StatusOK, allPariwisata)
+	return c.JSON(http.StatusOK, err)
```
Comment (geshtng) [inline pkg/delivery/pariwisata/http/pariwisata.go]: Ini kenapa kembaliannya err?

File: pkg/models/pariwisata.go [hunk 15]
```diff
@@ -7,7 +7,13 @@ package models
...
 type Pariwisata struct {
-	ID     int64
-	Name   string
-	Lokasi string
+	ID     int64  `json:"id"`
+	Name   string `json:"name"`
+	Lokasi string `json:"lokasi"`
+}
+
+type Respone struct {
```
Comment (geshtng) [inline pkg/models/pariwisata.go]:
```suggestion
type Response struct {
```

File: pkg/repository/pariwisata/postgres/pariwisata.go [hunk 16]
```diff
@@ -6,23 +6,66 @@ import (
...
+	MessageSucces = "Berhasil"
+	Messagefailed = "Gagal"
+
+	StatusOk     = "Ok"
+	StatusFailed = "Gagal"
+)
+
+func (r *pariwisata) GetAllPariwisata() (models.Respone, error) {
 	allPariwisata := []models.Pariwisata{}
+	var resp models.Respone
```
Comment (geshtng) [inline pkg/repository/pariwisata/postgres/pariwisata.go]:
```suggestion
	var resp models.Response
```

File: pkg/repository/pariwisata/postgres/pariwisata.go [hunk 17]
```diff
@@ -6,23 +6,66 @@ import (
 	"github.com/TobaTourism/pkg/models"
 )
 
-func (r *pariwisata) GetAllPariwisata() ([]models.Pariwisata, error) {
+const (
+	MessageSucces = "Berhasil"
+	Messagefailed = "Gagal"
+
+	StatusOk     = "Ok"
```
Comment (geshtng) [inline pkg/repository/pariwisata/postgres/pariwisata.go]:
```suggestion
	StatusOk     = "OK"
```

File: pkg/repository/pariwisata/postgres/pariwisata.go [hunk 18]
```diff
@@ -6,23 +6,66 @@ import (
...
+func (r *pariwisata) GetAllPariwisata() (models.Respone, error) {
 	allPariwisata := []models.Pariwisata{}
+	var resp models.Respone
 
-	rows, err := r.DB.Query(queryGetAllPariwisata)
+	rows, err := r.DB.Query(QueryGetAllPariwisata)
 	if err != nil {
-		log.Println("Repository error : ", err)
-		return allPariwisata, err
+		log.Println("[Repository][Experience][GetAllPariwisata] Query Error: ", err)
```
Comment (geshtng) [inline pkg/repository/pariwisata/postgres/pariwisata.go]:
```suggestion
		log.Println("[Repository][Pariwisata][GetAllPariwisata] Query Error: ", err)
```

File: pkg/repository/pariwisata/postgres/pariwisata.go [hunk 19]
```diff
@@ -6,23 +6,66 @@ import (
...
+	statement, err := r.DB.Prepare(queryCreatePariwisata)
+
+	if err != nil {
+		return err
+	}
+
+	defer statement.Close()
+
+	_, err = statement.Exec(pariwisataName, pariwisataLokasi)
+
```
Comment (geshtng) [inline pkg/repository/pariwisata/postgres/pariwisata.go:193]: Delete this line

File: pkg/repository/pariwisata/postgres/pariwisata.go [hunk 20]
```diff
@@ -6,23 +6,66 @@ import (
...
+	if err != nil {
+		return err
```
Comment (geshtng) [inline pkg/repository/pariwisata/postgres/pariwisata.go]: Logging

File: pkg/repository/pariwisata/postgres/pariwisata.go [hunk 21]
```diff
@@ -6,23 +6,66 @@ import (
...
+
+	return resp, nil
+}
+
+func (r *pariwisata) CreatePariwisata(pariwisataName string, pariwisataLokasi string) error {
+
+	statement, err := r.DB.Prepare(queryCreatePariwisata)
+
+	if err != nil {
+		return err
```
Comment (geshtng) [inline pkg/repository/pariwisata/postgres/pariwisata.go]: Logging

File: pkg/repository/pariwisata/postgres/statement.go [hunk 22]
```diff
@@ -1,3 +1,5 @@
 package postgres
 
-var queryGetAllPariwisata = "SELECT pariwisata_id, pariwisata_name, pariwisata_lokasi FROM pariwisata"
+var QueryGetAllPariwisata = "SELECT pariwisata_id, pariwisata_name, pariwisata_lokasi FROM pariwisata"
+
+var queryCreatePariwisata = `INSERT INTO pariwisata (pariwisata_name, pariwisata_lokasi)VALUES($1, $2) RETURNING pariwisata_id`
```
Comment (geshtng) [inline pkg/repository/pariwisata/postgres/statement.go]: Ganti ke const aja

File: pkg/usecase/pariwisata/module/pariwisata.go [hunk 23]
```diff
@@ -1,14 +1,42 @@
...
 import (
+
+	// "log"
+	"log"
+
 	"github.com/TobaTourism/pkg/models"
 )
 
-func (u *pariwisata) GetAllPariwisata() ([]models.Pariwisata, error) {
+func (u *pariwisata) GetAllPariwisata() (models.Respone, error) {
```
Comment (geshtng) [inline pkg/usecase/pariwisata/module/pariwisata.go]:
```suggestion
func (u *pariwisata) GetAllPariwisata() (models.Response, error) {
```

File: pkg/usecase/pariwisata/module/pariwisata.go [hunk 24]
```diff
@@ -1,14 +1,42 @@
...
+// 	// return allPariwisata, err
+// }
+
+func (u *pariwisata) CreatePariwisata(pName, pLokasi string) error {
+	// allPariwisata, err := u.pariwisataRepo.CreatePariwisata()
 	// if err != nil {
 	// 	log.Println(err)
 	// }
+	err := u.pariwisataRepo.CreatePariwisata(pName, pLokasi)
+
```
Comment (geshtng) [inline pkg/usecase/pariwisata/module/pariwisata.go:41]: Cek err != nil ???

File: pkg/usecase/pariwisata/usecase.go [hunk 25]
```diff
@@ -3,5 +3,6 @@ package pariwisata
 import "github.com/TobaTourism/pkg/models"
 
 type Usecase interface {
-	GetAllPariwisata() ([]models.Pariwisata, error)
+	GetAllPariwisata() (models.Respone, error)
```
Comment (geshtng) [inline pkg/usecase/pariwisata/usecase.go]:
```suggestion
	GetAllPariwisata() (models.Response, error)
```

File: pkg/repository/pariwisata/postgres/pariwisata.go [hunk 26]
```diff
@@ -6,23 +6,66 @@ import (
 	"github.com/TobaTourism/pkg/models"
 )
 
-func (r *pariwisata) GetAllPariwisata() ([]models.Pariwisata, error) {
+const (
+	MessageSucces = "Berhasil"
```
Comment (geshtng) [inline pkg/repository/pariwisata/postgres/pariwisata.go]:
```suggestion
	MessageSuccess = "Berhasil"
```

File: pkg/repository/pariwisata/postgres/pariwisata.go [hunk 27]
```diff
@@ -6,23 +6,66 @@ import (
...
+	Messagefailed = "Gagal"
+
+	StatusOk     = "Ok"
+	StatusFailed = "Gagal"
```
Comment (geshtng) [inline pkg/repository/pariwisata/postgres/pariwisata.go]:
```suggestion
	StatusFailed = "Failed"
```

File: pkg/repository/pariwisata/repository.go [hunk 28]
```diff
@@ -5,5 +5,6 @@ import (
 )
 
 type Repository interface {
-	GetAllPariwisata() ([]models.Pariwisata, error)
+	GetAllPariwisata() (models.Respone, error)
```
Comment (geshtng) [inline pkg/repository/pariwisata/repository.go]:
```suggestion
	GetAllPariwisata() (models.Response, error)
```

**Review by jhonsonsamueltua — APPROVED** [review]

**Pull request merged**